README.md

```diff
@@ -1,8 +1,68 @@
+# JSON Compilation Database Format Specification
+
+This document describes a format for specifying how to replay single link commands independently of the build system.
+
+## Background
+Currently, there is compilation database, which specified by [Clang](https://clang.llvm.org/docs/JSONCompilationDatabase.html). Compilation commands are actively used for syntax analysis of code, 
+but there are cases, when the compilation commands cannot provide all the necessary information, for example, look into simple C/C++ code:  
+`a.cpp:`
+```
+int f(int a) {
+  return g(a);
+}
+```
+`x.cpp:`
+```
+int g(int a) {
+  return a + 1;
+}
+```
+`y.cpp:`
+```
+int g(int a) {
+  return a - 1;
+}
 ```
-python3 main.py --i="path-to-project"
-``` 
+A tool may need to know which function is actually called, but this information is only available at linking step as soon as there may be several functions with same name in different units.  
+Using of links database is providing advanced software analysis.
+More examples:
 
-for information use ```python3 main.py -h```
- 
+1) Unit testing tools which use code analysis need to understand which function is called in order to get correct test result.
+2) IDEs need it to understand which file to go when you click "Go to definition" on a function call.
 
+## Supported Systems
 
+Currently CMake (since X.Y.Z) supports generation of link databases for Unix Makefile builds (Ninja builds in the works) with the option CMAKE_EXPORT_LINK_COMMANDS.
+
+There is no an alternative to intercept link calls with a other tool.
+
+## Format
+### Version 1.0.0
+Currently, there is no standard for the format of linking commands, so we providing version of format.
+
+A link database is a JSON file, which consist of an array of “command objects”, where each command object specifies one way a target or library.
```

Comment:
Library can also be a target.

Reply:
Please describe possible commands or targets: executable, shared library, static library, runlib command, other commands

Reply:
> Library can also be a target.

Anyway, It's not conflict with my statement

Reply:
Could you please remove "library" since it doesn't add any new information

Reply:
Please say something about runlib command and lwyu=...